pkg/build/graph/analysis/bc.go

```diff
@@ -0,0 +1,45 @@
+package analysis
+
+import (
+	"fmt"
+	"strings"
+
+	osgraph "github.com/openshift/origin/pkg/api/graph"
+	buildedges "github.com/openshift/origin/pkg/build/graph"
+	buildgraph "github.com/openshift/origin/pkg/build/graph/nodes"
+	imageedges "github.com/openshift/origin/pkg/image/graph"
+	imagegraph "github.com/openshift/origin/pkg/image/graph/nodes"
+)
+
+const (
+	MissingRequiredRegistryWarning = "MissingRequiredRegistry"
+)
+
+// FindUnpushableBuildConfigs checks all build configs that will output to an IST backed by an ImageStream and checks to make sure their builds can push.
+func FindUnpushableBuildConfigs(g osgraph.Graph) []osgraph.Marker {
+	bcNames := []string{}
+
+bc:
+	for _, bcNode := range g.NodesByKind(buildgraph.BuildConfigNodeKind) {
+		for _, istNode := range g.SuccessorNodesByEdgeKind(bcNode, buildedges.BuildOutputEdgeKind) {
+			for _, uncastImageStreamNode := range g.SuccessorNodesByEdgeKind(istNode, imageedges.ReferencedImageStreamGraphEdgeKind) {
+				imageStreamNode := uncastImageStreamNode.(*imagegraph.ImageStreamNode)
+				if len(imageStreamNode.Status.DockerImageRepository) == 0 {
+					bcNames = append(bcNames, bcNode.(*buildgraph.BuildConfigNode).ResourceString())
+					continue bc
+				}
+			}
+		}
+	}
+
+	if len(bcNames) == 0 {
+		return []osgraph.Marker{}
+	}
+
+	return []osgraph.Marker{{
```

Comment:
> I don't know what you're asking about the behavior - rephrase?

I have four build configs, three of them are unable to push because the openshift registry is not present.  Should I get three warnings or one warning?

```
Warnings:
  Some builds are pointing to image streams, but the administrator has not configured the integrated Docker registry: bc/one, bc/two, bc/three.  (oadm registry)
```
or
```
Warnings:
  bc/one cannot push to istag/one:latest because the administrator has not configured the integrated Docker registry.  (oadm registry)
  bc/two cannot push to istag/two:latest because the administrator has not configured the integrated Docker registry.  (oadm registry)
  bc/three cannot push to istag/three:latest because the administrator has not configured the integrated Docker registry.  (oadm registry)
```


The latter is a more pure use of the markers, the former is more terse.  My preference is for the three discrete markers.  I'd rather not have status trying to analyze warning types to selectively collapse them.